api/src/main/java/org/apache/cloudstack/api/command/user/network/ListNetworksCmd.java

```diff
@@ -101,7 +102,13 @@ public Long getZoneId() {
     }
 
     public String getGuestIpType() {
-        return guestIpType;
+        if (!Strings.isNullOrEmpty(guestIpType)) {
+            if (guestIpType.equalsIgnoreCase("all")) {
+                return null;
+            }
+            return Network.GuestType.fromValue(guestIpType).toString();
+        }
+        return null;
```

Comment:
so in the end empty and "all" are treated the same?

Reply:
that's right!